src/modules/endpoint/sagas/fetchResourceCache.js

```diff
@@ -1,31 +1,37 @@
 // @flow
 
 import type { Saga } from 'redux-saga'
-import { isEmpty, reduce } from 'lodash'
+import { flatten, isEmpty, reduce, values, mapValues, pickBy } from 'lodash'
 import { call, put } from 'redux-saga/effects'
 import type { ResourcesFetchFailedActionType } from '../../app/StoreActionType'
-import type { FetchResultType } from '../../fetcher/FetcherModule'
+import type { FetchResultType, TargetFilePathsType } from '../../fetcher/FetcherModule'
 import FetcherModule from '../../fetcher/FetcherModule'
-import { invertBy, mapValues, pickBy } from 'lodash/object'
 import type { DataContainer } from '../DataContainer'
+import type { FileCacheStateType } from '../../app/StateType'
 
-export type PathType = string
-export type UrlType = string
-export type FilePathType = string
-export type HashType = string
-export type FetchMapType = { [filePath: FilePathType]: [UrlType, PathType, HashType] }
+export type FetchMapTargetType = { url: string, filePath: string, urlHash: string }
+export type FetchMapEntryType = Array<FetchMapTargetType>
```

Comment:
You might remove this type, since it's probably less readable with it (too many types; my brain's buffer is limited :sweat_smile: )